test/vip/batch_address_test_tool/match_test.clj

```diff
@@ -0,0 +1,27 @@
+(ns vip.batch-address-test-tool.match-test
+  (:require [vip.batch-address-test-tool.match :refer :all]
+            [clojure.test :refer :all]))
+
+(deftest calculate-score-test
+  (testing "Exact match"
+    (is (= 0 (calculate-score {:expected-polling-location "foo"
+                               :api-result "foo"}))))
+  (testing "Off by 1"
+    (is (= 1 (calculate-score {:expected-polling-location "foo1"
+                               :api-result "foo"}))))
+  (testing "No API result"
+    (is (= -1 (calculate-score {:expected-polling-location "foo"})))))
+
+(deftest calculate-match-test
+  (testing "Match"
+    (dorun
+      (for [n (take 10 (iterate inc 0))]
+        (is (= "Match" (calculate-match {:score n}))))))
+  (testing "Possible Mismatch"
+    (dorun
+      (for [n (take 11 (iterate inc 10))]
+        (is (= "Possible Mismatch" (calculate-match {:score n}))))))
```

Comment:
I like this!  I think I'll implement the changes as suggested unless you have some strong objection @tank157.